### PR TITLE
Add notes

### PR DESCRIPTION
**Pull request by osamasayed**

### CODE REVIEW

File: types/NoteType.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+enum NoteType {
+  AyahsRange = 'ayahsRange',
```
Comment (osamasayed) [inline types/NoteType.ts:2]: let's remove this.

File: types/auth/NotesByTypeAndTypeIdResponse.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { BaseResponse } from '../ApiResponses';
+
+export type NoteResponse = {
+  id: string;
+  title: string;
+  body: string;
+  createdAt: Date;
+  updatedAt: Date;
```
Comment (osamasayed) [inline types/auth/NotesByTypeAndTypeIdResponse.ts:8]: We should add `ranges` field here

File: src/components/QuranReader/TranslationView/TranslationViewVerse/TranslationPageVerse.tsx [hunk 3]
```diff
@@ -51,6 +57,18 @@ const TranslationPageVerse: React.FC<TranslationPageVerse> = ({
     return response;
   });
 
+  const { data: notesCount } = useSWR(
```
Comment (osamasayed) [inline src/components/QuranReader/TranslationView/TranslationViewVerse/TranslationPageVerse.tsx:60]: Why cannot we just use `useSWRImmutable`?

> `useSWR(key, fetcher, { revalidateIfStale: false, revalidateOnFocus: false, revalidateOnReconnect: false })` is equivalent to `useSWRImmutable(key, fetcher)`

File: src/utils/auth/api.ts [hunk 4]
```diff
@@ -315,6 +320,46 @@ export const addCollection = async (collectionName: string) => {
...
+  // //   attachedEntityPayload = {
+  // //     attachedEntity: {
+  // //       entityType: type,
+  // //       entityId: typeId,
+  // //       ...(typeMetadata && {
+  // //         entityMetadata: typeMetadata,
+  // //       }),
+  // //     },
+  // //   };
+  // // }
```
Comment (osamasayed) [inline src/utils/auth/api.ts:343]: we should remove this

File: src/utils/auth/api.ts [hunk 5]
```diff
@@ -315,6 +320,46 @@ export const addCollection = async (collectionName: string) => {
...
+
+  const payload = {
+    title,
+    body,
+    ranges,
+    // ...attachedEntityPayload,
```
Comment (osamasayed) [inline src/utils/auth/api.ts:349]: ditto: we should remove this

File: src/components/Verse/Notes/index.tsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+const VerseNotes = ({
+  verseKey,
+  isTranslationView,
+  trigger = VerseNotesTrigger.PopoverItem,
+}: VerseNotesProps) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const { t } = useTranslation('common');
+
+  const onItemClicked = () => {
+    logButtonClick('verse_actions_menu_note', {
```
Comment (osamasayed) [inline src/components/Verse/Notes/index.tsx:34]: nit: the trigger type should be included here

File: src/components/Notes/NoteModal/utils/ranges.ts
Comment (ahmedriad1) [inline src/components/Notes/NoteModal/utils/ranges.ts:1]: Remove this util, we already have `parseVerseRange`

File: src/components/Notes/NoteModal/NoteRangesIndicator/index.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    return <></>;
+  }
+
+  // TODO: this is temporary and assumes that a note has only one range and 1 Ayah inside that range
+  const [verseKey, chapterNumber] = parseNoteRanges(ranges);
+  const chapterData = getChapterData(chaptersData, chapterNumber.toString());
+  const verseKeyName = `${chapterData.transliteratedName} ${toLocalizedVerseKey(
+    verseKey.toString(),
+    lang,
+  )}`;
```
Comment (ahmedriad1) [inline src/components/Notes/NoteModal/NoteRangesIndicator/index.tsx:30]: Remove this code and do this:
```ts
const [
  { chapter: fromChapter, verse: fromVerse, verseKey: rangeFrom },
] = parseVerseRange(range);
const verseKeyName = `${getChapterData(chaptersData, fromChapter).transliteratedName} ${toLocalizedNumber(Number(fromVerse), lang)}`;
```

**Review by ahmedriad1 — CHANGES_REQUESTED** [review]

**Pull request merged**